lib/resty/limit/req.lua

```diff
@@ -160,17 +163,34 @@ function _M.limit(cfg)
 
     rate = floor((tonumber(rate) or 1) * 1000 / scale)
 
-    local res, err = redis_lookup(conn, zone, key, rate, interval, burst)
+    local res, err = redis_lookup(conn_slave, zone, key, rate, interval, burst)
     if res and (res[1] == _M.BUSY or res[1] == _M.FORBIDDEN) then
         if res[1] == _M.BUSY then
-            ngx.log(log_level, 'limiting requests, excess ' ..
-                        res[2]/1000 .. ' by zone "' .. zone .. '"')
+            ngx.log(log_level, "excess requests " ..
+                    zone .. ":" .. key .. " - " .. res[2]/1000 )
         end
         return
     end
-
     if not res and err then
-        ngx.log(ngx.WARN, "redis lookup err: ", err)
+        ngx.log(ngx.WARN, "redis lookup error: ", err)
+        return _M.OK
+    end
+
+    for host, conn in pairs(conn_masters) do
+        local _, err = conn:ping()
+        if err then
+            ngx.log(ngx.WARN, "failed to ping redis master: " .. host .. " ", err)
+            return _M.OK
+        end
+    end
+
+    for host, conn in pairs(conn_masters) do
+      local res, err = redis_update(conn, zone, key, res[2], interval, burst)
+      if err then
+          ngx.log(ngx.WARN, "failed to update redis master host: " .. host ..
```

Comment:
`failed to update redis master host` --> `failed to update redis master` (as above)